fees/rocketpool.ts

```diff
@@ -0,0 +1,174 @@
+import { Interface, ZeroAddress } from "ethers";
+import { normalizeAddress } from "@defillama/sdk/build/util";
+
+/**
+ * 
+ * There are two main actors in rocketpool system: stakers and node operators
+ * 
+ * Stakers stake any amount of ETH into rETH - these ETH will be borrowed and staking into the Beacon chain by node operators
+ * Node operators open minipools by stake atleast 8 or 16 ETH and must borrow 24 or 16 ETH from rETH and operate a validator on the Beacon chain
+ * 
+ * When node operators earn ETH rewards from Beacon chain, they earn 5%-14% commission
+ * And node operators distribute remaining ETH rewards amount to rETH stakers
+ * 
+ * Here is an example scenario:
+ * 
+ * Node A open open a minipool of 8 ETH and with a commission of 5%, it need to borrow 24 ETH from rETH
+ * Node A then stakes all 32 ETH and become a active validator on the Beacon chain
+ * Node A earns 0.5 ETH rewards from Beacon chain and will distribute as follow:
+ * - Total 5% of 0.5 ETH (0.025) earned by Node A as commission
+ * - Node A also earned 25% if remaining 0.475 ETH from initial 8 ETH staking in minipool
+ * - Node A distributes all remaing of 75% of 0.475 ETH to rETh stakers
+ * 
+ * As descrive above, total fees on Rocketpool are distributed to supply side stakers and minipools operators.
+ * The Rocket Pool protocol matches liquid stakers with node operators. It does not take a fee for providing this service.
+ */
+
+const methodology = {
+  Fees: 'Total ETH staking rewards from Rocketpool active validators',
+  SupplySideRevenue: 'Total ETH staking rewards are distributed to rETH stakers and minipool depositors.',
+}
+
+const rETH = '0xae78736cd615f374d3085123a210448e74fc6393';
+
+// MEV and priority fees send to smoothing pool by block builders
+const smoothingPool = '0xd4E96eF8eee8678dBFf4d535E033Ed1a4F7605b7';
+
+const RocketPoolContractAbis = {
+  rETHEventEtherDeposited: 'event EtherDeposited(address indexed from, uint256 amount, uint256 time)',
+
+  // count ETH withdrawn from smoothing pool as collected fees
+  smoothingPoolEtherWithdrawn: 'event EtherWithdrawn(string indexed by, address indexed to, uint256 amount, uint256 time)',
+
+  // these call target to RocketMinipool
+  // but abi from RocketMinipoolDelegate
+  // https://etherscan.io/address/0xA347C391bc8f740CAbA37672157c8aAcD08Ac567#code
+  minipoolGetNodeFee: 'function getNodeFee() view returns (uint256)',
+  minipoolGetNodeDepositBalance: 'function getNodeDepositBalance() view returns (uint256)',
+}
+
+interface EtherDepositedEvent {
+  blockNumber: number;
+  etherAmount: bigint;
+  fromAddress: string;
+}
+
+async function fetch(options: FetchOptions): Promise<FetchResultV2> {
+  const dailyFees = options.createBalances()
+
+  const rETHContract: Interface = new Interface([
+    RocketPoolContractAbis.rETHEventEtherDeposited,
+  ])
+  const etherDepositedEvents: Array<EtherDepositedEvent> = (await options.getLogs({
+    target: rETH,
+    entireLog: true,
```

Comment:
same question here, I thought the blockNumber was used, else fetch the parsed logs here, cleaner code?